domain_config_ui/domain_config_ui.services.yml

```diff
@@ -0,0 +1,16 @@
+services:
+  domain_config_ui.manager:
+    class: Drupal\domain_config_ui\DomainConfigUIManager
+    arguments: ['@config.storage', '@domain.loader', '@language_manager']
```

Comment:
Was there a specific code reason that made doing so necessary? 

Ideally, we'd handle all of the negotiation inside the DomainConfigOverrider, or, if needed, create a DomainConfigManager service for this.

My thought here is that the UI piece should not handle any of the negotiation logic, just the form handling, so that it could be swapped.